src/Sylius/Bundle/ApiBundle/Validator/Constraints/EligibleProductVariantToCartValidator.php

```diff
@@ -14,59 +14,74 @@
 namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
 
 use Sylius\Bundle\ApiBundle\Command\Cart\AddItemToCart;
+use Sylius\Bundle\ApiBundle\Command\Cart\ChangeItemQuantityInCart;
 use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
 use Sylius\Component\Core\Model\ProductInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
 use Sylius\Component\Core\Repository\OrderRepositoryInterface;
 use Sylius\Component\Inventory\Checker\AvailabilityCheckerInterface;
 use Sylius\Component\Product\Repository\ProductVariantRepositoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
 use Symfony\Component\Validator\Constraint;
 use Symfony\Component\Validator\ConstraintValidator;
 use Webmozart\Assert\Assert;
 
 /** @experimental */
-final class AddingEligibleProductVariantToCartValidator extends ConstraintValidator
+final class EligibleProductVariantToCartValidator extends ConstraintValidator
 {
     /** @var ProductVariantRepositoryInterface */
     private $productVariantRepository;
 
+    /** @var RepositoryInterface */
+    private $orderItemRepository;
+
     /** @var OrderRepositoryInterface */
     private $orderRepository;
 
     /** @var AvailabilityCheckerInterface  */
     private $availabilityChecker;
 
+    /** @var ProductVariantInterface|null */
+    private $productVariant;
+
+    /** @var string */
+    private $productVariantCode = '';
+
     public function __construct(
         ProductVariantRepositoryInterface $productVariantRepository,
+        RepositoryInterface $orderItemRepository,
         OrderRepositoryInterface $orderRepository,
         AvailabilityCheckerInterface $availabilityChecker
     ) {
         $this->productVariantRepository = $productVariantRepository;
+        $this->orderItemRepository = $orderItemRepository;
         $this->orderRepository = $orderRepository;
         $this->availabilityChecker = $availabilityChecker;
     }
 
     public function validate($value, Constraint $constraint)
     {
-        Assert::isInstanceOf($value, AddItemToCart::class);
+        if (!$value instanceof ChangeItemQuantityInCart && !$value instanceof AddItemToCart) {
+            throw new \InvalidArgumentException;
+        }
```

Comment:
```suggestion
        Assert::isInstanceOf($value, [AddItemToCart::class, ChangeItemQuantityInCart::class]);
```

Reply:
I checked it, only object/string is possible
<img width="1096" alt="Screenshot 2021-05-20 at 13 36 23" src="https://user-images.githubusercontent.com/39232096/118972086-737d4f80-b970-11eb-8e03-e19b757b6ede.png">